src/app/loadout-builder/process.ts

```diff
@@ -250,20 +246,23 @@ export function process(
               ];
 
               const firstValidSet = getFirstValidSet(armor);
-              const statChoices = [
-                keyToStats(helmsKey),
-                keyToStats(gauntsKey),
-                keyToStats(chestsKey),
-                keyToStats(legsKey),
-                keyToStats(classItemsKey),
-                keyToStats(ghostsKey)
-              ];
               if (firstValidSet) {
+                const statChoices = [
+                  keyToStats(helmsKey),
+                  keyToStats(gauntsKey),
+                  keyToStats(chestsKey),
+                  keyToStats(legsKey),
+                  keyToStats(classItemsKey),
+                  keyToStats(ghostsKey)
+                ];
+
                 const maxPower = getPower(firstValidSet);
+
+                const stats = {};
                 for (const stat of statChoices) {
                   let index = 0;
                   for (const key of statKeys) {
-                    stats[key] += stat[index];
+                    stats[key] = (stat[key] || 0) + stat[index];
```

Comment:
I think this should be `stats[key] = (stats[key] || 0) + stat[index];`, it creating a few problems atm.

Reply:
You're right.